diff -ur TGICL-2.1.orig/bin/tgicl TGICL-2.1/bin/tgicl
--- TGICL-2.1.orig/bin/tgicl	2019-03-20 13:16:04.000000000 +0100
+++ TGICL-2.1/bin/tgicl	2019-03-20 13:18:10.000000000 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/usr/bin/env perl
 use strict;
 use Getopt::Std;
 use IO::File;
diff -ur TGICL-2.1.orig/bin/tgicl_asm.psx TGICL-2.1/bin/tgicl_asm.psx
--- TGICL-2.1.orig/bin/tgicl_asm.psx	2019-03-20 13:16:04.000000000 +0100
+++ TGICL-2.1/bin/tgicl_asm.psx	2019-03-20 13:18:50.000000000 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/usr/bin/env perl
 use strict;
 use FindBin;
 umask 0002;
diff -ur TGICL-2.1.orig/bin/tgicl_asmpta.psx TGICL-2.1/bin/tgicl_asmpta.psx
--- TGICL-2.1.orig/bin/tgicl_asmpta.psx	2019-03-20 13:16:04.000000000 +0100
+++ TGICL-2.1/bin/tgicl_asmpta.psx	2019-03-20 13:19:11.000000000 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/usr/bin/env perl
 use strict;
 use FindBin;
 umask 0002;
